django/db/models/expressions.py

```diff
@@ -287,7 +287,13 @@ def _resolve_output_field(self):
         sources_iter = (source for source in self.get_source_fields() if source is not None)
         for output_field in sources_iter:
             if any(not isinstance(output_field, source.__class__) for source in sources_iter):
```

Comment:
```suggestion
            source = None
            if any(not isinstance(output_field, source.__class__) for source in sources_iter):
```
What about this @felixxm

Reply:
`source` will be `None`.

Reply:
`source` will be **always** `None` so it is not a proper solution.